cli/flox/src/commands/activate.rs

```diff
@@ -375,118 +399,48 @@ impl Activate {
-    /// to put the system curated path items first in the `PATH`,
-    /// which completely breaks the user's ability to manage their `PATH` in subshells,
-    /// e.g. when using tmux.
-    ///
-    /// On macos `/usr/libexec/path_helper` is typically invoked from
-    /// default shell rc files, e.g. `/etc/profile` and `/etc/zprofile`.
-    ///
-    /// Note: since the "path_helper" i only invoked by login shells,
-    /// we only need to setup the PATH patching for `flox activate` in shell rc files.
-    ///
-    /// ## Example
-    ///
-    /// > User has `eval "$(flox activate)"` in their `.zshrc`.
-    ///
-    ///  Without the path patching, the following happens:
-    ///
-    /// 1. Open a new terminal (login shell)
-    ///     -> `path_helper` runs (`PATH=<default envs>`)
-    ///     -> `flox activate` runs (`PATH=<flox_env>:<default envs>`)
-    /// 2. Open a new tmux session (login shell by default)
-    ///     -> `path_helper` runs (`PATH=<default envs>:<flox_env>`)
-    ///     -> `flox activate` runs
-    ///     -> ⚡️ environment already active, activate skipped
-    ///        without path patching: `PATH:<default envs>:<flox_env>` ❌
-    ///        with path patching: `PATH:<flox_env>:<default envs>`    ✅ flox env is not shadowed
-    ///
-    /// ## Implementation
-    ///
-    /// This function attempts to undo the effects of `/usr/libexec/path_helper`
-    /// by partitioning the `PATH` into two parts:
-    /// 1. known paths of activated flox environments
-    ///    and nix store paths (e.g. `/nix/store/...`) -- assumed to be `nix develop` paths
-    /// 2. everything else
+    /// Activate the environment interactively by spawning a new shell
+    /// and running the respective activation scripts.
     ///
-    /// The `PATH` is then reordered to put the flox environment and nix store paths first.
-    /// The order within the two partitions is preserved.
-    #[cfg_attr(not(target_os = "macos"), allow(unused_variables))] // on linux `flox_env_dirs` is not used
-    fn fixup_path(flox_env_dirs: &IndexSet<PathBuf>) -> Option<Result<OsString>> {
-        #[cfg(not(target_os = "macos"))]
-        {
-            None
-        }
-        #[cfg(target_os = "macos")]
-        {
-            if !Path::new("/usr/libexec/path_helper").exists() {
-                return None;
-            }
-            let path_var = env::var("PATH").unwrap_or_default();
-            let fixed_up_path = Self::fixup_path_with(path_var, flox_env_dirs);
-            let fixed_up_path_joined = env::join_paths(fixed_up_path).context(
-                "Cannot activate environment because its path contains an invalid character",
-            );
-            Some(fixed_up_path_joined)
+    /// This function should never return as it replaces the current process
+    fn activate_interactive(
+        shell: Shell,
+        exports: HashMap<&str, String>,
+        activation_path: PathBuf,
```

Comment:
```suggestion
activation_path: &impl AsRef<Path>
```
We use `PathBuf` for this function but `&Path` for another. I think in all cases we should prefer `&impl AsRef<Path>` as this allows you to provide any type that can be converted into a `&Path`.

Reply:
In the body of the function you just call `activation_path.as_ref()` to do the conversion to `&Path`.

Reply:
Style, can address in followup